jax/scipy/sparse/linalg.py

```diff
@@ -17,68 +17,89 @@
 
 import numpy as np
 import jax.numpy as jnp
+from jax import scipy as jsp
 from jax import lax, device_put
-from jax.tree_util import tree_leaves, tree_map, tree_multimap, tree_structure
+from jax.tree_util import (tree_leaves, tree_map, tree_multimap, tree_structure,
+                           tree_reduce, Partial)
 from jax.util import safe_map as map
 
 
+_dot = partial(jnp.dot, precision=lax.Precision.HIGHEST)
+_vdot = partial(jnp.vdot, precision=lax.Precision.HIGHEST)
+
+# aliases for working with pytreedef _vdot_real_part(x, y):
+
 def _vdot_real_part(x, y):
-  """Vector dot-product guaranteed to have a real valued result."""
+  """Vector dot-product guaranteed to have a real valued result despite
+     possibly complex input. Thus neglects the real-imaginary cross-terms.
+     The result is a real float.
+  """
   # all our uses of vdot() in CG are for computing an operator of the form
-  # `z^T M z` where `M` is positive definite and Hermitian, so the result is
+  #  z^T M z
```

Comment:
This is the same as z^H M z = conj(z)^H M z, which is the natural inner product for complex-valued vector spaces.

Reply:
I think the T here was meant to denote Hermitian conjugation, but I'll change the docstring